docs/WalletGuide.md

```diff
@@ -0,0 +1,301 @@
+  assetCode,
+  authToken,
+});
+```
+
+Get a single transaction's current status and details.
+
+```typescript
+const transaction = await anchor.getTransactionBy({
+  authToken,
+  id: transactionId,
+});
+```
+
+Get account transactions for a specified asset.
+
+```typescript
+const transactions = await anchor.getHistory({
+  authToken,
+  assetCode,
+});
+```
+
+Watch transaction.
+
+```typescript
+let watcher = anchor.watcher();
+
+let { stop } = watcher.watchOneTransaction({
+  authToken,
+  assetCode,
+  id: successfulTransaction.id,
+  onMessage,
+  onSuccess,
+  onError,
+});
+
+// stop watching transaction
+stop();
+```
+
+### Add client domain signing
+
+One of the features being supported
+by [SEP-10 Authentication](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md)
+is verifying client domain. This enables anchor to recognize that user request was made using a specific client.
+
+Functionality of signing with client domain is enabled by the [WalletSigner](../wallet-sdk/src/walletsdk/Auth/WalletSigner.ts). You can use the `DefaultSigner` class as a baseline for your implementation. The main method that needs to be implemented is `signWithDomainAccount`.
+
+Implementation should make a request to a backend server containing the client domain private key. Server returns signed transaction, finishing the flow.
+
+For your wallet signer to be used, it's required to set it in this SDK. It's recommended to set it globally via `ApplicationConfiguration` `defaultSigner` [option](../wallet-sdk/src/walletsdk/index.ts).
+Alternatively, it can be passed to the `authenticate` call of the `Auth` class.
+
+#### Example
+
+In the example below, [Stellar demo wallet](https://demo-wallet.stellar.org) will be used as a client domain. Server-side implementation, responsible for signing transaction can be found [here](https://github.com/stellar/stellar-demo-wallet/tree/master/packages/demo-wallet-server).  
+
+First, create an object using the `WalletSigner` interface, and define both the signWithClientAccount and a signWithDomainAccount method.
+
+```typescript
+const demoWalletSigner: WalletSigner = {
+  signWithClientAccount: ({ transaction, accountKp }) => {
+    transaction.sign(accountKp);
+    return transaction;
+  },
+  signWithDomainAccount: async ({
+    transactionXDR,
+    networkPassphrase,
+    accountKp,
+  }) => {
+    return await axios.post("https://demo-wallet-server.stellar.org/sign", {
+      transactionXDR,
+      networkPassphrase,
+    });
+  },
+};
+```
+
+Wallet can now use this class:
+```typescript
+const wallet = new walletSdk.Wallet({
+  stellarConfiguration: walletSdk.StellarConfiguration.TestNet(),
+  applicationConfiguration: new walletSdk.ApplicationConfiguration(
+    demoWalletSigner
+  ),
+});
+```
+
+And it can now be used for authentication with client domain:
+```typescript
+const authToken = await auth.authenticate({
+  accountKp,
+  walletSigner: demoWalletSigner,
+  memoId: "",
```

Comment:
Why is it empty?

Reply:
it doesn't have to be. removed
https://github.com/stellar/typescript-wallet-sdk/pull/25/commits/eda70f3cad4ea9f6590b748a19509aacdc4d7764